core/templates/pages/story-editor-page/editor-tab/story-editor.component.spec.ts

```diff
@@ -0,0 +1,532 @@
+    component.updateStoryThumbnailBgColor('abcd');
+    expect(storyUpdateSpy).toHaveBeenCalled();
+  });
+
+  it('should return the classroom and topic url fragment', () => {
+    expect(component.getClassroomUrlFragment()).toEqual('math');
+    expect(component.getTopicUrlFragment()).toEqual('fractions');
+  });
+
+  it('should not open confirm or cancel modal if the initial node is' +
+      ' being deleted',
+  () => {
+    let modalSpy = spyOn(ngbModal, 'open');
+    component.deleteNode('node_2');
+    expect(modalSpy).not.toHaveBeenCalled();
+  });
+
+  it('should open confirm or cancel modal when a node is being deleted',
+    fakeAsync(() => {
+      let modalSpy = spyOn(ngbModal, 'open').and.returnValue({
+        result: Promise.resolve()
+      } as NgbModalRef);
+      let storyUpdateSpy = spyOn(
+        storyUpdateService, 'deleteStoryNode').and.stub();
+
+      component.deleteNode('node_1');
+      tick();
+
+      expect(storyUpdateSpy).toHaveBeenCalled();
+      expect(modalSpy).toHaveBeenCalled();
+    }));
+
+  it('should call storyUpdateService to add destination node id',
+    () => {
+      class MockComponentInstance {
+        compoenentInstance!: {
+          nodeTitles: null;
+        };
+      }
+
+      let modalSpy = spyOn(ngbModal, 'open').and.callFake(() => {
+        return ({
+          componentInstance: MockComponentInstance,
+          result: Promise.resolve()
+        }) as NgbModalRef;
+      });
+
+      component.createNode();
+
+      expect(modalSpy).toHaveBeenCalled();
+    });
+
+  it('should call storyUpdateService to add destination node id',
+    fakeAsync(() => {
+      class MockComponentInstance {
+        compoenentInstance!: {
+          nodeTitles: null;
+        };
+      }
+
+      let sampleStoryBackendObject = {
+        id: 'sample_story_id',
+        title: 'Story title',
+        description: 'Story description',
+        notes: 'Story notes',
+        version: 1,
+        corresponding_topic_id: 'topic_id',
+        story_contents: {
+          initial_node_id: 'node_1',
+          nodes: [
+            {
+              id: 'node_1',
+              title: 'Title 1',
+              description: 'Description 1',
+              prerequisite_skill_ids: ['skill_1'],
+              acquired_skill_ids: ['skill_2'],
+              destination_node_ids: [],
+              outline: 'Outline',
+              exploration_id: 'exp_id',
+              outline_is_finalized: false,
+              thumbnail_filename: 'fileName',
+              thumbnail_bg_color: 'blue',
+            }],
+          next_node_id: 'node_1'
+        },
+        language_code: 'en',
+        thumbnail_filename: 'fileName',
+        thumbnail_bg_color: 'blue',
+        url_fragment: 'url',
+        meta_tag_content: 'meta'
+      };
+
+      spyOn(component, '_initEditor').and.stub();
+      component.story = Story.createFromBackendDict(
+        sampleStoryBackendObject);
+
+      tick();
```

Comment:
Can you add a comment explaining why a `tick` is required here?

Reply:
removed